### PR TITLE
feat: Specify the different entry file rather than index.md

### PR DESCRIPTION
**Pull request by g4rry420**
Signed-off-by: GurkiranSingh <gurkiransinghk@gmail.com>

User can specify different entry rather `index.md` by using the `entry` config option. Please, let me know if you see any issues :)
Resolves: #280

### CODE REVIEW

File: src/node/serve/serve.ts [hunk 1]
```diff
@@ -31,9 +31,17 @@ export async function serve(options: ServeOptions = {}) {
   const notAnAsset = (pathname: string) => !pathname.includes('/assets/')
   const notFound = fs.readFileSync(path.resolve(site.outDir, './404.html'))
   const onNoMatch: IOptions['onNoMatch'] = (req, res) => {
-    res.statusCode = 404
-    if (notAnAsset(req.path)) res.write(notFound.toString())
-    res.end()
+    if (site.site.entry !== 'index' && req.path === '/') {
```
Comment (brc-dd) [inline src/node/serve/serve.ts:34]: Yeah but this logic won't work when someone is deploying their site. It only works with our custom server which is only suitable for previewing the website locally after build. The servers specifically need `index.html` to be present. That being said, we need to build `entry` as if it was `index.md` (and emit `index.html`).

VuePress has probably done this using their permalink feature, which is still missing in VitePress.

Comment (g4rry420) [reply src/node/serve/serve.ts:34]: I agree, I was worried by this too. I will look into VuePress implementation for this.

File: src/node/plugin.ts [hunk 2]
```diff
@@ -153,6 +154,8 @@ export async function createVitePressPlugin(
           id,
           config.publicDir
         )
+        // console.log('id -- ', id)
+        // console.log('deadLinks -- ', deadLinks)
         if (deadLinks.length) {
```
Comment (g4rry420) [inline src/node/plugin.ts:159]:
```suggestion
          config.publicDir
        )
        if (deadLinks.length) {
```

**Comment by g4rry420** [comment]
@brc-dd If user specifies the different `entry` file other than `index`, then I am emitting `index.html` file, however, the entry file js chunk remained the same name, which shouldn't be a problem as we are only importing it in the index file.
Also, if user wants to add link to the `entry` page, then they will have to point to `index` file only, otherwise it will result in deadlinks.
Please, let me know if you see any issues :)

**Pull request closed without merging**